src/v/archival/ntp_archiver_service.cc

```diff
@@ -759,10 +759,22 @@ ss::future<> ntp_archiver::maybe_flush_delta_offset_next() {
     // Translate just past the end, and persist the delta. This value is
     // required to report the high watermark (one offset past the last
     // consumable offset).
-    auto next_offset = model::next_offset(manifest.get_last_offset());
     auto ot_state = _parent.get_offset_translator_state();
-    auto delta_next = model::offset_delta(
-      next_offset() - ot_state->from_log_offset(next_offset)());
+    model::offset_delta delta_next;
+    try {
+        delta_next = ot_state->delta_next_offset(manifest.get_last_offset());
```

Comment:
nit: This belongs in the previous commit

Reply:
Rejiggered this PR